### PR TITLE
Support test and filter plugins

### PR DESCRIPTION
**Pull request by felixfontein**
Also improves support for lookup plugins by handling the special option `_terms`.

Tests will fail at the moment.

Result:
- Filter plugins: https://ansible.fontein.de/collections/ansible/builtin/#filter-plugins
- Test plugins: https://ansible.fontein.de/collections/ansible/builtin/#test-plugins
- Lookup plugin example: https://ansible.fontein.de/collections/ansible/builtin/first_found_lookup.html#terms

### CODE REVIEW

**Comment by briantist** [comment]
Ah! I didn't realize this wasn't done/published yet. Explains why my docs are not showing it and by references are wrong 😅 
https://github.com/ansible-collections/community.hashi_vault/pull/263

Any guess on when this will be released (approximately)?

**Comment by felixfontein** [comment]
Let's discuss this at Tuesday's DaWGs meeting. (I.e. definitely not before then :) )

File: src/antsibull_docs/data/docsite/plugin.rst.j2 [hunk 1]
```diff
@@ -162,17 +162,90 @@ The below requirements are needed on the local controller node that executes thi
...
+
+.. Positional
+
+Positional parameters
+---------------------
+
+{%   if plugin_type == 'filter' %}
+This describes positional parameters of the filter. This are the values ``positional1``, ``positional2`` and so on in the following example: ``input | @{plugin_name}@(positional1, positional2, ...)``.
+{%   elif plugin_type == 'test' %}
+This describes positional parameters of the test. This are the values ``positional1``, ``positional2`` and so on in the following examples: ``input is @{plugin_name}@(positional1, positional2, ...)`` and ``input is not @{plugin_name}@(positional1, positional2, ...)``.
```
Comment (briantist) [inline src/antsibull_docs/data/docsite/plugin.rst.j2:216]:
```suggestion
This describes positional parameters of the test. These are the values ``positional1``, ``positional2`` and so on in the following examples: ``input is @{plugin_name}@(positional1, positional2, ...)`` and ``input is not @{plugin_name}@(positional1, positional2, ...)``.
```

File: src/antsibull_docs/data/docsite/plugin.rst.j2 [hunk 2]
```diff
@@ -162,17 +162,90 @@ The below requirements are needed on the local controller node that executes thi
...
+------------------
+{%   else %}
 Parameters
 ----------
+{%   endif %}
+
+{%   if plugin_type == 'filter' %}
+This describes keyword parameters of the filter. This are the values ``key1=value1``, ``key2=value2`` and so on in the following example: ``input | @{plugin_name}@(key1=value1, key2=value2, ...)``.
+{%   elif plugin_type == 'test' %}
+This describes keyword parameters of the test. This are the values ``key1=value1``, ``key2=value2`` and so on in the following examples: ``input is @{plugin_name}@(key1=value1, key2=value2, ...)`` and ``input is not @{plugin_name}@(key1=value1, key2=value2, ...)``.
```
Comment (briantist) [inline src/antsibull_docs/data/docsite/plugin.rst.j2:242]:
```suggestion
This describes keyword parameters of the test. These are the values ``key1=value1``, ``key2=value2`` and so on in the following examples: ``input is @{plugin_name}@(key1=value1, key2=value2, ...)`` and ``input is not @{plugin_name}@(key1=value1, key2=value2, ...)``.
```

**Comment by felixfontein** [comment]
@briantist @samccann @acozine thanks for reviewing and commenting on the output :)

**Pull request merged**